### ISSUE TITLE
21.11.0 crashes in MigrationHelper.java

### ISSUE THREAD

**Issue by marko-petrovic** [Countly/countly-sdk-android#140]
When upgrading Android Countly SDK from `20.04.2` to `21.11.0` (or 21.11.1) and after "Countly.sharedInstance().init" there is a crash in `MigrationHelper.java`, when `performMigration0To1()` is called.   
  
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference.  
Line 115, as line 110 sets `deviceIDType` to be null.  

This is my initialization    
```
val countlyConfig = CountlyConfig(
            application,
            appKey,
            serverUrl
        )
            .enablePublicKeyPinning(
                    arrayOf("xyz")
             )
            .setDeviceId(
                    "myDeviceId"
             )
            .enableCrashReporting()
            .setRecordAllThreadsWithCrash()
            .setLoggingEnabled(true)

Countly.sharedInstance().init(countlyConfig)
```
adding `setIdMode` before calling init doesn't help.

**Comment by ArtursKadikis** [comment]
Thank you for letting us know. I'll release a fix to address this in the coming days.

**Comment by ArtursKadikis** [comment]
Hello, we just published the 21.11.2 version. It should fix this issue. Let me know if you still encounter the issue.
